File: remove-linked-list-elements/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        p1 = head
        prev = None
        while p1:
            if p1.val == val:
                if p1 == head: #Still at head
                    p1 = p1.next
                    head = head.next
                else: #Not at head
                    prev.next = p1.next
                    p1 = p1.next
            else: #Move forward
                prev = p1
                p1 = p1.next
        return head

File: remove-linked-list-elements/remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution 
{
    public ListNode removeElements(ListNode head, int target) 
    {
        if (head == null) // List does not exist
            return head;
        
        ListNode it = new ListNode(-1, head);
        head = it;
        while (it.next != null)
        {
            if (it.next.val == target)
                it.next = it.next.next;
            else
                it = it.next;
        }
        return head.next;
    }
}